flink-runtime/src/main/java/org/apache/flink/runtime/operators/DataSourceTask.java

```diff
@@ -75,6 +78,13 @@
 	// cancel flag
 	private volatile boolean taskCanceled = false;
 
+	/**
+	 * The accumulator map used in the RuntimeContext.
+	 */
+	protected Map<String, Accumulator<?,?>> accumulatorMap;
```

Comment:
This field should be final. Or you get rid of the field and just access the accumulators via `getEnvironment().getAccumulatorRegistry().getUserMap()`.